python3.8/cookiecutter-aws-sam-eventBridge-python/{{cookiecutter.project_name}}/tests/integration/test_ec2_event.py

```diff
@@ -0,0 +1,118 @@
+import logging
+import os
+from time import sleep, time
+from unittest import TestCase
+
+import boto3
+from botocore.exceptions import ClientError
+
+"""
+Make sure env variable AWS_SAM_STACK_NAME exists with the name of the stack we are going to test. 
+"""
+
+
+class TestEC2Event(TestCase):
+    """
+    This integration test will create an EC2 and verify the lambda function is invoked by checking the cloudwatch log
+    The EC2 instance will be deleted when test completes.
+    """
+
+    function_name: str
+    instance_id: str  # temporary EC2 instance ID
+
+    @classmethod
+    def get_and_verify_stack_name(cls) -> str:
+        stack_name = os.environ.get("AWS_SAM_STACK_NAME")
+        if not stack_name:
+            raise Exception(
+                "Cannot find env var AWS_SAM_STACK_NAME. \n"
+                "Please setup this environment variable with the stack name where we are running integration tests."
+            )
+
+        # Verify stack exists
+        client = boto3.client("cloudformation")
+        try:
+            client.describe_stacks(StackName=stack_name)
+        except Exception as e:
+            raise Exception(
+                f"Cannot find stack {stack_name}. \n" f'Please make sure stack with the name "{stack_name}" exists.'
+            ) from e
+
+        return stack_name
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        stack_name = TestEC2Event.get_and_verify_stack_name()
+
+        client = boto3.client("cloudformation")
+        response = client.list_stack_resources(StackName=stack_name)
+        resources = response["StackResourceSummaries"]
+        function_resources = [
+            resource for resource in resources if resource["LogicalResourceId"] == "HelloWorldFunction"
+        ]
+        if not function_resources:
+            raise Exception("Cannot find HelloWorldFunction")
+
+        cls.function_name = function_resources[0]["PhysicalResourceId"]
+
+    def setUp(self) -> None:
+        client = boto3.client("ec2")
+        response = client.run_instances(
+            ImageId="resolve:ssm:/aws/service/ami-amazon-linux-latest/amzn2-ami-hvm-x86_64-gp2",
+            InstanceType="t2.nano",
+            MaxCount=1,
+            MinCount=1,
+        )
+        self.assertIn("Instances", response, "Fail to create an EC2 instance")
+        self.instance_id = response["Instances"][0]["InstanceId"]
+
+    def tearDown(self) -> None:
+        client = boto3.client("ec2")
+        client.terminate_instances(InstanceIds=[self.instance_id])
+
+    def test_ec2_event(self):
+        log_group_name = f"/aws/lambda/{self.function_name}"
+
+        client = boto3.client("logs")
+
+        # cloudwatch log might be deplayed, give it 5 retries
+        retries = 5
+        start_time = int(time() - 60) * 1000
```

Comment:
nit: Python allows masking this sort of logic really nicely with decorators, which would allow writing e.g.

```python
@retry(5)
def test_ec2_event(self):
    ...
```

Just something to keep in mind if this sort of logic is likely to be repeated.